test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2626,6 +2626,36 @@ public void PackageBuilderWorksWithFileNameWithoutAnExtension()
             }
         }
 
+        [Theory]
+        [InlineData(".txt", "The 'icon' element 'icon.txt' has an invalid file extension. Valid options are .png, .jpg or .jpeg.")]
+        [InlineData(".jpeg", null)]
+        [InlineData(".jpg", null)]
+        [InlineData(".png", null)]
+        [InlineData(".PnG", null)]
+        [InlineData(".jPG", null)]
+        [InlineData(".jpEG", null)]
+        [InlineData("", "The 'icon' element 'icon' has an invalid file extension. Valid options are .png, .jpg or .jpeg.")]
+        public void Icon_IconInvalidExtension_ThrowsException(string fileExtension, string message)
```

Comment:
Most of the inline data appears to be testing that no exception is thrown, contradicting the name of the test.

Reply:
I will separate failing and succeeding tests